apps/frontend/src/components/TimeCardPage/EvaluationModal.tsx

```diff
@@ -0,0 +1,182 @@
+import {
+  Stack,
+  Button,
+  Modal,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  FormLabel,
+  RadioGroup,
+  Radio,
+  Input,
+} from "@chakra-ui/react";
+import React from "react";
+
+import { useState, useEffect } from "react";
+import { CommentSchema } from "src/schemas/RowSchema";
+import { createNewComment } from "./utils";
+import { UserSchema } from "../../schemas/UserSchema";
+import { CommentType } from "./types";
+
+export function EvaluationModal({
+  isOpen,
+  onClose,
+  onCommentSubmit,
+  onCommentDelete,
+  commentToEdit,
+}: {
+  isOpen: boolean;
+  onClose: () => void;
+  onCommentSubmit: (comment: CommentSchema) => void;
+  onCommentDelete: (comment: CommentSchema) => void;
+  commentToEdit: CommentSchema | null;
+}) {
+  const [editedComment, setEditedComment] = useState<string>(
+    commentToEdit ? commentToEdit.Content : ""
+  );
+  const [isOpenCommentForm, setIsOpenCommentForm] = useState(false);
+  const [selectedOption, setSelectedOption] = useState("");
+  const [otherText, setOtherText] = useState("");
+  const [commentList, setCommentList] = useState<CommentSchema[]>([]);
+  const [selectedComment, setSelectedComment] = useState<CommentSchema | null>(
+    null
+  );
+
+  interface WeeklyEvaluationModalProps {
+    setWeeklyEvaluation: Function;
+    weeklyEvaluation: CommentSchema[];
+  }
+
+  // set value for existing comment
+  useEffect(() => {
+    if (commentToEdit) {
+      setEditedComment(commentToEdit.Content);
+    } else {
+      setEditedComment("");
+    }
+  }, [commentToEdit]);
+
+  const options = [
+    {
+      value: "option1",
+      content: "This associate was a total rock star this week!",
+    },
+    { value: "option2", content: "This associate did just okay this week" },
+    {
+      value: "option3",
+      content: "Lack of professionalism and/or poor behavior",
+    },
+    { value: "option4", content: "N/A" },
+    { value: "option5", content: otherText },
+  ];
+
+  const [user, setUser] = useState<UserSchema>();
+
+  const closeCommentForm = () => {
+    setIsOpenCommentForm(false);
+    onClose();
+  };
+
+  const handleOptionChange = (value: string) => {
```

Comment:
It might be nice for some of these functions to give a quick single line comment about what they are actually doing. I know they're pretty simple for the most part, but good practice.